lib/galaxy/tools/__init__.py

```diff
@@ -825,6 +830,44 @@ def tests(self):
             self.__tests_populated = True
         return self.__tests
 
+    @property
+    def _repository_dir(self):
+        """If tool shed installed tool, the base directory of the repository installed."""
+        repository_dir = None
+
+        if hasattr(self, 'tool_shed') and self.tool_shed:
+            repository_dir = self.tool_dir
+            while True:
+                repository_dir_name = os.path.basename(repository_dir)
+                if repository_dir_name == self.repository_name:
+                    break
+
+                parent_repository_dir = os.path.dirname(repository_dir)
+                if repository_dir == parent_repository_dir:
+                    log.error("Problem finding repository dir for tool [%s]" % self.id)
+                    repository_dir = None
+
+        return repository_dir
+
+    def test_data_path(self, filename):
+        repository_dir = self._repository_dir
+        if repository_dir:
+            for root, dirs, files in os.walk(repository_dir):
+                if '.' in dirs:
```

Comment:
`if '.hg' in dirs:`

Reply:
Huh - this was copied and pasted but doesn't make a lot of sense.

```diff
-        for root, dirs, files in os.walk(os.path.join(tool_path, installed_tool_path)):
-            if '.' in dirs:
-                dirs.remove('.hg')
```

 I think you must be correct about the original intent. I'll update with that fix.